internal/controller/pdbwatcher_controller.go

```diff
@@ -41,28 +44,44 @@ func (r *PDBWatcherReconciler) Reconcile(ctx context.Context, req ctrl.Request)
 	err := r.Get(ctx, req.NamespacedName, pdbWatcher)
 	if err != nil {
 		if errors.IsNotFound(err) {
-			return ctrl.Result{}, nil
+			return ctrl.Result{}, nil // PDBWatcher not found, could be deleted, nothing to do
+		}
+		return ctrl.Result{}, err // Error fetching PDBWatcher
+	}
+
+	// Check for conflicts with other PDBWatchers
```

Comment:
Curious what made you tackle this in this particular pr. 

Another way to handle this is to hve the pdbwatch match the name of hte pdb instead of bing in the spec.  This also works but is a little bit expensive and they both fail if ther are duplicates.

Reply:
But probbly fine.